src/vs/workbench/contrib/tasks/test/common/taskConfiguration.test.ts

```diff
@@ -1760,3 +1765,107 @@ suite('Bugs / regression tests', () => {
 		testConfiguration(external, builder);
 	});
 });
+
+class TestNamedProblemMatcher implements Partial<ProblemMatcher> {
+}
+
+class TestParseContext implements Partial<ParseContext> {
+}
+
+suite('To task configuration from', () => {
+	let instantiationService: TestInstantiationService;
+	let parseContext: ParseContext;
+	let namedProblemMatcher: NamedProblemMatcher;
+	let problemReporter: ProblemReporter;
+	setup(() => {
+		instantiationService = new TestInstantiationService();
+		namedProblemMatcher = instantiationService.createInstance(TestNamedProblemMatcher);
+		namedProblemMatcher.name = 'real';
+		namedProblemMatcher.label = 'real label';
+		problemReporter = new ProblemReporter();
+		parseContext = instantiationService.createInstance(TestParseContext);
+		parseContext.problemReporter = problemReporter;
+		parseContext.namedProblemMatchers = { 'real': namedProblemMatcher };
+		parseContext.uuidMap = new UUIDMap();
+	});
+	suite('ProblemMatcher config', () => {
```

Comment:
nit: I'd normally use a method name here to use existing terminology and make it more searchable:

```suggestion
	suite('ProblemMatcherConverter.from', () => {
```

Reply:
outdated, but updated